src/EventStore.Core/ClusterVNode.cs

```diff
@@ -526,6 +532,26 @@ public class ClusterVNode :
 					Db.Config.WriterCheckpoint.AsReadOnly(),
 					optimizeReadSideCache: Db.Config.OptimizeReadSideCache));
 
+			var streamInfoCacheCapacity = options.Cluster.StreamInfoCacheCapacity;
+			//qq thoughts
+			// 1. maybe now the default should be 0 (or even -1 to indicate dynamic or something)
+			// 2. i wonder if we should base the calculation off of total system memory rather than free memory, not sure.
+			// 3. there is some chance of GetFreeMem returning -1, we might want a different value to 50k in that case
+			// 4. probably best stick this as a method in another class for some easy unit testing
+			if (streamInfoCacheCapacity == Opts.StreamInfoCacheCapacityDefault) {
+				var availableMem = statsHelper.GetFreeMem();
+				if (availableMem > 2L * 1024 * 1024 * 1024) {
+					// capacity uses apporixmately 1kib per stream (including both caches)
+					// so we want capacity * 1024 = availableMem
+					ulong bytesPerUnitCapacity = 1024;
+					streamInfoCacheCapacity = (int)(availableMem / bytesPerUnitCapacity);
+				} else {
+					streamInfoCacheCapacity = 50_000;
+				}
+				Log.Debug("Set StreamInfoCacheCapacity to {streamInfoCacheCapacity}. Calculated based on {availableMem} bytes of free memory.",
```

Comment:
```suggestion
				Log.Information("Set StreamInfoCacheCapacity to {streamInfoCacheCapacity}. Calculated based on {availableMem} bytes of free memory.",
```
This log can be useful for diagnosing issues